airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_incremental.py

```diff
@@ -76,9 +78,33 @@ def records_with_state(records, state, stream_mapping, state_cursor_paths) -> It
         yield record_value, state_value, stream_name
 
 
+def compare_cursor_with_threshold(record_value, state_value, threshold_days) -> bool:
+    """
+    Checks if the record's cursor value is older or equal to the state cursor value.
+
+    If the threshold_days option is set, the values will be converted to dates so that the time-based offset can be applied.
+    :raises dateutil.parser._parser.ParserError: if threshold_days is passed with non-date cursor values.
+    """
+    if threshold_days:
+        record_date_value = record_value if isinstance(record_value, datetime) else dateutil.parser.parse(record_value)
+        state_date_value = state_value if isinstance(state_value, datetime) else dateutil.parser.parse(state_value)
+
+        return record_date_value >= (state_date_value - timedelta(days=threshold_days))
+
+    return record_value >= state_value
```

Comment:
This is only considering the `threshold_days` option to be valid for date-based cursors. An offset may be desired for other types of cursors (maybe ints?), but I couldn't think of a good use case and it would make the parsing and comparison more difficult.